server/src/main/scala/controllers/application/ShiftsController.scala

```diff
@@ -160,7 +148,43 @@ class ShiftsController @Inject()(cc: ControllerComponents,
   def saveShifts: Action[AnyContent] = Action.async { request =>
     request.body.asText.map { text =>
       val newShifts = shiftsFromJson(text)
-      createShifts(newShifts)
+      if (newShifts.size == 1) {
```

Comment:
This isn't quite right mate. It means that if we change the UI in the future to allow multiple additional shifts to be created in one go then this end point will treat them as the first shifts and skip the checks.

What we could do with here is extracting the code to add a single shift and map over the shifts to call it for each one. 

We shouldn't need the `createShifts` function unless it's doing things that the add shift code doesn't do.

Is there a reason why this code can't be used to add the first shifts?